src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -117,6 +119,29 @@ namespace winrt::TerminalApp::implementation
         _filteredActionsView().ScrollIntoView(_filteredActionsView().SelectedItem());
     }
 
+    // Method Description:
+    // - Called when the command selection changes. We'll use this in the tab
+    //   switcher to "preview" tabs as the user navigates the list of tabs. To
+    //   do that, we'll dispatch the switch to tab command for this tab, but not
+    //   dismiss the switcher.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void CommandPalette::_selectedCommandChanged(const IInspectable& /*sender*/,
+                                                 const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
+    {
+        if (_currentMode == CommandPaletteMode::TabSwitchMode)
+        {
+            const auto& selectedCommand = _filteredActionsView().SelectedItem();
+            if (const auto& command = selectedCommand.try_as<TerminalApp::Command>())
+            {
+                const auto& actionAndArgs = command.Action();
+                _dispatch.DoAction(actionAndArgs);
+            }
+        }
+    }
```

Comment:
This seems circuitous. TerminalPage is already driving the selection of tabs in the tab switcher, so  should we reduce coupling by having TerminalPage do the active terminal change in addition to driving the switching UI?

Maybe having it be in MRU order changes things.

Reply:
(I'm averse to taking on too many more "are we in mode A, B, C, or D" checks :/)